### ISSUE TITLE
Sporadic timeout failure of listStream unit test

### ISSUE THREAD

**Issue by Abhijeet-Jadhav** [pravega/pravega#1004]
**Problem description**
ListStream timeout failure is seen on local and travis builds.
```
com.emc.pravega.controller.rest.v1.StreamMetaDataTests > testListStreams FAILED
    org.junit.runners.model.TestTimedOutException: test timed out after 5 seconds
        at java.net.SocketInputStream.socketRead0(Native Method)
        at java.net.SocketInputStream.socketRead(SocketInputStream.java:116)
        at java.net.SocketInputStream.read(SocketInputStream.java:170)
        at java.net.SocketInputStream.read(SocketInputStream.java:141)
        at java.io.BufferedInputStream.fill(BufferedInputStream.java:246)
        at java.io.BufferedInputStream.read1(BufferedInputStream.java:286)
        at java.io.BufferedInputStream.read(BufferedInputStream.java:345)
        at sun.net.www.MeteredStream.read(MeteredStream.java:134)
        at java.io.FilterInputStream.read(FilterInputStream.java:133)
        at sun.net.www.protocol.http.HttpURLConnection$HttpInputStream.read(HttpURLConnection.java:3375)
        at org.glassfish.jersey.client.internal.HttpUrlConnector$2.read(HttpUrlConnector.java:228)
        at org.glassfish.jersey.message.internal.EntityInputStream.read(EntityInputStream.java:102)
        at org.glassfish.jersey.message.internal.ReaderInterceptorExecutor$UnCloseableInputStream.read(ReaderInterceptorExecutor.java:296)
        at com.fasterxml.jackson.core.json.ByteSourceJsonBootstrapper.ensureLoaded(ByteSourceJsonBootstrapper.java:522)
        at com.fasterxml.jackson.core.json.ByteSourceJsonBootstrapper.detectEncoding(ByteSourceJsonBootstrapper.java:129)
        at com.fasterxml.jackson.core.json.ByteSourceJsonBootstrapper.constructParser(ByteSourceJsonBootstrapper.java:246)
        at com.fasterxml.jackson.core.JsonFactory._createParser(JsonFactory.java:1271)
        at com.fasterxml.jackson.core.JsonFactory.createParser(JsonFactory.java:810)
        at com.fasterxml.jackson.jaxrs.base.ProviderBase._createParser(ProviderBase.java:822)
        at com.fasterxml.jackson.jaxrs.base.ProviderBase.readFrom(ProviderBase.java:763)
        at org.glassfish.jersey.message.internal.ReaderInterceptorExecutor$TerminalReaderInterceptor.invokeReadFrom(ReaderInterceptorExecutor.java:256)
        at org.glassfish.jersey.message.internal.ReaderInterceptorExecutor$TerminalReaderInterceptor.aroundReadFrom(ReaderInterceptorExecutor.java:235)
        at org.glassfish.jersey.message.internal.ReaderInterceptorExecutor.proceed(ReaderInterceptorExecutor.java:155)
        at org.glassfish.jersey.message.internal.MessageBodyFactory.readFrom(MessageBodyFactory.java:1085)
        at org.glassfish.jersey.message.internal.InboundMessageContext.readEntity(InboundMessageContext.java:874)
        at org.glassfish.jersey.message.internal.InboundMessageContext.readEntity(InboundMessageContext.java:808)
        at org.glassfish.jersey.client.ClientResponse.readEntity(ClientResponse.java:326)
        at org.glassfish.jersey.client.InboundJaxrsResponse$1.call(InboundJaxrsResponse.java:115)
        at org.glassfish.jersey.internal.Errors.process(Errors.java:315)
        at org.glassfish.jersey.internal.Errors.process(Errors.java:297)
        at org.glassfish.jersey.internal.Errors.process(Errors.java:228)
        at org.glassfish.jersey.process.internal.RequestScope.runInScope(RequestScope.java:419)
        at org.glassfish.jersey.client.InboundJaxrsResponse.runInScopeIfPossible(InboundJaxrsResponse.java:267)
        at org.glassfish.jersey.client.InboundJaxrsResponse.readEntity(InboundJaxrsResponse.java:112)
        at com.emc.pravega.controller.rest.v1.StreamMetaDataTests.testListStreams(StreamMetaDataTests.java:562)
```

**Problem location**
REST unit test.

**Suggestions for an improvement**
Increase timeout to 20 seconds.

**Comment by Abhijeet-Jadhav** [comment]
Observed this issue: https://travis-ci.org/pravega/pravega/builds/240003119?utm_source=github_status&utm_medium=notification

**Comment by Abhijeet-Jadhav** [comment]
Observed this again: https://github.com/pravega/pravega/issues/1501

**Comment by skrishnappa** [comment]
This should be fixed with PR - #1527, where a similar issue was observed and fixed by buffering the response in the client before reading. 
Please re-open if observed again.
